yarn-project/simulator/src/public/tail_phase_manager.ts

```diff
@@ -115,4 +127,35 @@ export class TailPhaseManager extends AbstractPhaseManager {
       N
     >;
   }
+
+  private sortLogsHashes<N extends number>(unencryptedLogsHashes: Tuple<SideEffect, N>): Tuple<SideEffect, N> {
+    return sortByCounter(unencryptedLogsHashes.map(n => ({ ...n, counter: n.counter.toNumber() }))).map(
+      h => new SideEffect(h.value, new Fr(h.counter)),
+    ) as Tuple<SideEffect, N>;
+  }
+
+  // As above, this is a hack for unencrypted logs ordering, now they are sorted. Since the public kernel
+  // cannot keep track of side effects that happen after or before a nested call, we override the gathered logs.
+  // As a sanity check, we at least verify that the elements are the same, so we are only tweaking their ordering.
+  // See same fn in pxe_service.ts
+  // Added as part of resolving #5017
+  private patchLogsOrdering(tx: Tx, publicInputs: PublicKernelCircuitPublicInputs) {
+    const unencLogs = tx.unencryptedLogs.unrollLogs();
+    const sortedUnencLogs = publicInputs.end.unencryptedLogsHashes;
+
+    const finalUnencLogs: UnencryptedL2Log[] = [];
+    sortedUnencLogs.forEach((sideEffect: SideEffect) => {
+      if (!sideEffect.isEmpty()) {
+        const isLog = (log: UnencryptedL2Log) => Fr.fromBuffer(log.hash()).equals(sideEffect.value);
+        const thisLogIndex = unencLogs.findIndex(isLog);
+        finalUnencLogs.push(unencLogs[thisLogIndex]);
+      }
+    });
+    const unencryptedLogs = new UnencryptedFunctionL2Logs(finalUnencLogs);
+
+    tx.unencryptedLogs.functionLogs[0] = unencryptedLogs;
+    for (let i = 1; i < tx.unencryptedLogs.functionLogs.length; i++) {
+      tx.unencryptedLogs.functionLogs[i] = UnencryptedFunctionL2Logs.empty();
+    }
+  }
```

Comment:
These fns were added to mimic `sortNoteHashes` as we don't currently sort in public tail.